src/app/profiles/profiles.default.ts

```diff
@@ -0,0 +1,116 @@
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+
+import { Component } from '@angular/core';
+import { RMM } from '../rmm';
+import { RunboxWebmailAPI } from '../rmmapi/rbwebmail';
+import { timeout, share, retry } from 'rxjs/operators';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+    selector: 'app-profiles-default',
+    styleUrls: ['profiles.default.scss'],
+    templateUrl: 'profiles.default.html',
+})
+export class DefaultProfileComponent {
+    field_errors: any;
+    profiles: any[] = new Array();
+    selected: any;
+
+    constructor(public rmm: RMM, public rmmapi: RunboxWebmailAPI, private snackBar: MatSnackBar) {
+        this.selectCurrentDefault();
+    }
+
+    async fetchProfiles() {
+        const froms = await this.rmmapi.getFromAddress().toPromise();
+        // Sort emails alphabetically
+        this.profiles = froms.sort((a, b) => (a.email < b.email ? -1 : 1));
+    }
+
+    async selectCurrentDefault() {
+        await this.fetchProfiles();
+        const defaultProfiles = [];
+        for (const profile of this.profiles) {
+            if (profile.priority === '0' || profile.priority === 0) {
+                defaultProfiles.push(profile);
+            }
+        }
+        if (defaultProfiles.length === 1) {
+            this.selected = defaultProfiles[0];
+        } else {
+            for (const profile of defaultProfiles) {
+                if (profile.type === 'main') {
+                    this.selected = profile;
+                }
+            }
+        }
+    }
+
+    updateDefaultProfile() {
+        const priorities: any[] = new Array();
+        const priority_data = { from_priorities: priorities };
+        const type_data = { type: 'main' };
+        for (const profile of this.profiles) {
+            if (profile === this.selected) {
+                const values = { id: profile.id, from_priority: 0 };
+                priorities.push(values);
+                this.updateType(profile.id, type_data, this.field_errors);
+            } else {
+                const values = { id: profile.id, from_priority: 1 };
+                priorities.push(values);
+            }
+        }
+        this.updateFromPriorities(priority_data);
+    }
+
+    updateFromPriorities(values: { from_priorities: any[] }) {
+        const req = this.rmm.ua.http.put('/rest/v1/profile/from_priority/', values).pipe(retry(3), timeout(60000), share());
```

Comment:
I just mindlessly copied the `.pipe()` from another place in the app, because that's how it's used everywhere, with the 60s but I didn't think it through when adding the `.retry()`. Changed it to 10s instead.